src/azure-cli/azure/cli/command_modules/vm/operations/snapshot.py

```diff
@@ -0,0 +1,59 @@
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+# pylint: disable=no-self-use, line-too-long, protected-access, too-few-public-methods, unused-argument
+from knack.log import get_logger
+from knack.util import CLIError
+
+from azure.cli.core.aaz import has_value
+from ..aaz.latest.snapshot import Update as _SnapshotUpdate
+
+logger = get_logger(__name__)
+
+
+class SnapshotUpdate(_SnapshotUpdate):
+    @classmethod
+    def _build_arguments_schema(cls, *args, **kwargs):
+        from azure.cli.core.aaz import AAZStrArg
+        args_schema = super()._build_arguments_schema(*args, **kwargs)
+
+        args_schema.disk_access_id._registered = False
+        args_schema.disk_encryption_set_id._registered = False
+
+        args_schema.disk_access = AAZStrArg(
+            options=["--disk-access"],
+            help="Name or ID of the disk access resource for using private endpoints on disks.",
+        )
+        args_schema.disk_encryption_set = AAZStrArg(
+            options=["--disk-encryption-set"],
+            help="Name or ID of disk encryption set that is used to encrypt the disk."
+        )
+
+        return args_schema
+
+    def pre_instance_update(self, instance):
+        from azure.mgmt.core.tools import resource_id, is_valid_resource_id
+        from azure.cli.core.commands.client_factory import get_subscription_id
+
+        args = self.ctx.args
+        if has_value(args.disk_encryption_set):
+            if instance.properties.encryption.type != 'EncryptionAtRestWithCustomerKey' and \
+                    has_value(args.encryption_type) and \
+                    args.encryption_type != 'EncryptionAtRestWithCustomerKey':
+                raise CLIError('usage error: Please set --encryption-type to EncryptionAtRestWithCustomerKey')
```

Comment:
Could we use specific error type instead od CLIError?

Reply:
It is not new logic for this PR and just moved in from custom.py file.
But you are right, I'll change it to `ArgumentUsageError`.